### PR TITLE
Add UserAuth containers

### PR DESCRIPTION
**Pull request by keharper**
Adds documentation about UserAuth containers.

### CODE REVIEW

File: src/content/docs/dropins/user-auth/containers/auth-combine.mdx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import Diagram from '@components/Diagram.astro';
+import Callouts from '@components/Callouts.astro';
+import OptionsTable from '@components/OptionsTable.astro';
+import Aside from '@components/Aside.astro';
+import { Steps } from '@astrojs/starlight/components';
+
+The AuthCombine container renders a dialog that combines the sign-in, sign-up, and reset password forms. The user can switch between the forms using the provided links.
+
+<Diagram caption="AuthCombine container">
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/auth-combine.mdx:14]: Recommend removing placeholder image

File: src/content/docs/dropins/user-auth/containers/auth-combine.mdx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+The AuthCombine container provides the following configuration options:
+
+<OptionsTable
+compact
+    options={[
+        ['Options', 'Type', 'Req?', 'Description'],
+        ['defaultView', 'activeComponentType', 'No', 'Specifies which form is the default.'],
+        ['signInFormConfig', 'SignInFormProps', 'No', 'Renders the SignIn container.'],
+        ['signUpFormConfig', 'SignUpFormProps', 'No', 'Renders the SignUp container.'],
+        ['resetPasswordFormConfig', 'SignUpFormProps', 'No', 'Renters the ResetPassword container.'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/auth-combine.mdx]:
```suggestion
        ['resetPasswordFormConfig', 'SignUpFormProps', 'No', 'Renders the ResetPassword container.'],
```

File: astro.config.mjs [hunk 3]
```diff
@@ -185,6 +185,18 @@ export default defineConfig({
             collapsed: true,
             items: [
               { label: 'Overview', link: '/dropins/user-auth/' },
+              { label: 'Containers', 
+                collapsed: true,
+                items: [
+                  { label: 'AuthCombine', link: '/dropins/user-auth/containers/auth-combine/' },
```
Comment (jeff-matthews) [inline astro.config.mjs:191]: Should the list of containers be alphabetized?

File: src/content/docs/dropins/user-auth/containers/auth-combine.mdx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+---
+title: AuthCombine Container
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/auth-combine.mdx]:
```suggestion
title: AuthCombine container
```

Comment (jeff-matthews) [reply src/content/docs/dropins/user-auth/containers/auth-combine.mdx]: I think we tend to prefer sentence case for topic titles, right? If so, all topic titles need updates.

File: src/content/docs/dropins/user-auth/containers/sign-in.mdx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+description: Learn about the SignIn container in the UserAuth dropin.
+---
+
+import Diagram from '@components/Diagram.astro';
+import Callouts from '@components/Callouts.astro';
+import OptionsTable from '@components/OptionsTable.astro';
+import Aside from '@components/Aside.astro';
+import { Steps } from '@astrojs/starlight/components';
+
+The SignIn container helps render different screens that enable the customer to log in to an acount. The container supports multiple scenarios, including standard logins for existing accounts and for accounts that are process of being confirmed. If customers are required to confirm an email address to create an account, the **Stores** > Configuration > **Customers** > **Customer Configuration** > **Create New Account Options** > **Require Emails Confirmation** setting in the Commerce Admin must be set to **Yes**.
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-in.mdx]: Consider breaking this into two paragraphs. The default styling enlarges the first paragraph similar to the devsite repos. When the first paragraph is large, it looks odd on the rendered page.

File: src/content/docs/dropins/user-auth/containers/sign-in.mdx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+<Diagram caption="SignIn standard login container">
+  ![SignIn standard login container](@images/user-auth/sign-in-container.png)
+</Diagram>
+
+## Log in to an unconfirmed account
+
+This scenario is applicable when email confirmation has been enabled, and the user has created an account, but has not confirmed their email.
+
+If the user attempts to log in with credentials for an unconfirmed account,they will see a custom notification allowing them to resend the confirmation email. When the user clicks on the **Resend confirmation email** link shown below, Commerce sends a new confirmation email to the user.
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-in.mdx]:
```suggestion
If the user attempts to log in with credentials for an unconfirmed account, they will see a custom notification allowing them to resend the confirmation email. When the user clicks on the **Resend confirmation email** link shown below, Commerce sends a new confirmation email to the user.
```

File: src/content/docs/dropins/user-auth/containers/sign-in.mdx [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+<Diagram caption="SignIn standard login container">
+  ![SignIn standard login container](@images/user-auth/sign-in-not-confirmed.png)
+</Diagram>
+
+## Confirmation from email
+
+The SignIn container handles an email confirmation when the user clicks on the confirmation link they received via email. The link format is:
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-in.mdx]:
```suggestion
The SignIn container processes email confirmations when users click on the confirmation link sent to their email. The link format is:
```

File: src/content/docs/dropins/user-auth/containers/sign-in.mdx [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+`https://mcstaging.aemshop.net/customer/account/confirm/?back_url=&email=<user_email>&id=<user_id>&key=<confirmation_key>`
+
+The following steps describe the flow:
+
+* Email confirmation is enabled in the admin panel.
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-in.mdx]:
```suggestion
* Email confirmation is enabled in the Commerce Admin.
```

File: src/content/docs/dropins/user-auth/containers/sign-in.mdx [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+* The user has created an account and received the email confirmation link.
+* The user clicks on the link and gets redirected to the email confirmation page with the SignIn drop-in.
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-in.mdx]:
```suggestion
* The user clicks on the link and gets redirected to the email confirmation page with the SignIn dropin.
```

File: src/content/docs/dropins/user-auth/containers/sign-in.mdx [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+<Aside>
+To enable the email confirmation functionality, it is required to pass the `enableEmailConfirmation` prop with a true value during integration.
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-in.mdx]:
```suggestion
To enable the email confirmation functionality, you must pass the `enableEmailConfirmation` property with a `true` value during integration.
```

File: src/content/docs/dropins/user-auth/containers/sign-in.mdx [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+    ['formSize', 'default | small', 'No', 'Controls form paddings and spacing. Use "small" to embed the form in small layout containers like a dropdown in the header.'],
+    ['renderSignUpLink', 'boolean', 'No', 'Controls the visibility of the “Sign up” link next to the “Forgot Password” link.'],
+    ['initialEmailValue', 'string', 'No', 'Allows passing an initial value for the email input (prefilled sign-in form).'],
+    ['enableEmailConfirmation', 'boolean', 'No', 'Determines if this container is used to handle email confirmation.'],
+    ['hideCloseBtnOnEmailConfirmation', 'boolean', 'No', 'Controls the visibility of the “Close” button on the email confirmation view.'],
+    ['routeRedirectOnEmailConfirmationClose', 'function', 'No', 'Determines where the user is redirected when the “Close” button on the email confirmation view is clicked.'],
+    ['routeForgotPassword', 'function', 'No', 'Determines where the “Forgot password?” link redirects the customer.'],
+    ['routeSignUp', 'function', 'No', 'Determines where the “Sign up” link redirects the customer (link visibility is based on the renderSignUpLink prop).'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-in.mdx]:
```suggestion
    ['routeSignUp', 'function', 'No', 'Determines where the “Sign up” link redirects the customer (link visibility is based on the renderSignUpLink property).'],
```

File: src/content/docs/dropins/user-auth/containers/sign-in.mdx [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+    ['routeRedirectOnSignIn', 'function', 'No', 'Determines the page to which the user should be redirected after sign-in. If provided, the user won’t be redirected to the  SuccessNotification container or any slot-defined override.'],
+    ['onSuccessCallback', 'function', 'No', 'Callback executed when the user successfully logs in, receiving userName and status as parameters.'],
+    ['onErrorCallback', 'function', 'No', 'Callback executed when an error occurs, receiving the error object as a parameter.'],
+    ['onSignUpLinkClick', 'function', 'No', 'Executed on clicking the “Sign up” link before redirect (link visibility is based on the renderSignUpLink prop).'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-in.mdx]:
```suggestion
    ['onSignUpLinkClick', 'function', 'No', 'Executed on clicking the “Sign up” link before redirect (link visibility is based on the renderSignUpLink property).'],
```

File: src/content/docs/dropins/user-auth/containers/sign-up.mdx [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+description: Learn about the SignIn container in the UserAuth dropin.
+---
+
+import Diagram from '@components/Diagram.astro';
+import Callouts from '@components/Callouts.astro';
+import OptionsTable from '@components/OptionsTable.astro';
+import Aside from '@components/Aside.astro';
+import { Steps } from '@astrojs/starlight/components';
+
+The SignUp container provides a form where the user enter the details needed to create a new account. The execution of this form can vary, depending on how Commerce is configured:
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-up.mdx]:
```suggestion
The SignUp container provides a form where the user enters the details required to create a new account. The execution of this form can vary, depending on how Commerce is configured:
```

File: src/content/docs/dropins/user-auth/containers/sign-up.mdx [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+
+## SignUp configurations
+
+The SignUp container provides the following configuration options:
+
+<OptionsTable
+    compact
+    options={[
+        ['Options', 'Type', 'Req?', 'Description'],
+        ['addressesData', 'AddressFormProps[]', 'No', 'Allows passing address data into the SignUp drop-in. On successful sign-up, corresponding addresses will be automatically created in the newly created account.'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-up.mdx]:
```suggestion
        ['addressesData', 'AddressFormProps[]', 'No', 'Allows passing address data into the SignUp dropin. On successful sign-up, corresponding addresses will be automatically created in the newly created account.'],
```

File: src/content/docs/dropins/user-auth/containers/sign-up.mdx [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+        ['inputsDefaultValueSet', 'inputsDefaultValueSetProps[]', 'No', ' Allows passing default values for inputs (prefilled sign-up form).'],
+        ['fieldsConfigForApiVersion1', 'any', 'No', 'Allows passing a set of field configurations for environments that do not support the createCustomerV2 mutation (fallback, should not be used in environments with the latest AC version).'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-up.mdx]:
```suggestion
        ['fieldsConfigForApiVersion1', 'any', 'No', 'Allows passing a set of field configurations for environments that do not support the createCustomerV2 mutation (fallback, should not be used in environments with the latest Commerce version).'],
```

File: src/content/docs/dropins/user-auth/containers/sign-up.mdx [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+        ['apiVersion2', 'boolean', 'No', 'Defaults to true. Allows switching to the createCustomer mutation if createCustomerV2 is not supported in the current version of AC (fallback, should not be used in environments with the latest AC version).'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-up.mdx]:
```suggestion
        ['apiVersion2', 'boolean', 'No', 'Defaults to true. Allows switching to the createCustomer mutation if createCustomerV2 is not supported in the current version of Commerce (fallback, should not be used in environments with the latest Commerce version).'],
```

File: src/content/docs/dropins/user-auth/containers/sign-up.mdx [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+        ['displayTermsOfUseCheckbox', 'boolean', 'No', 'Controls displaying of the appropriate checkboxes in the sign-up form. Note: Rendered checkboxes are not functional and should not be used yet. They will be used in the future.'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-up.mdx]:
```suggestion
        ['displayTermsOfUseCheckbox', 'boolean', 'No', 'Controls displaying the appropriate checkboxes in the sign-up form. Rendered checkboxes are not functional and should not be used yet. They will be used in the future.'],
```

File: src/content/docs/dropins/user-auth/containers/sign-up.mdx [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+        ['displayNewsletterCheckbox', 'boolean', 'No', ''],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-up.mdx]: The description here is blank.

File: src/content/docs/dropins/user-auth/containers/sign-up.mdx [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+        ['isAutoSignInEnabled', 'boolean', 'No', 'Determines if the user should be automatically signed in after registration. This has no effect if email confirmation is enabled in the admin panel (automatic sign-in is not possible in this case).'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-up.mdx]:
```suggestion
        ['isAutoSignInEnabled', 'boolean', 'No', 'Determines if the user should be automatically signed in after registration. This has no effect if email confirmation is enabled in the Commerce Admin (automatic sign-in is not possible in this case).'],
```

File: src/content/docs/dropins/user-auth/containers/sign-up.mdx [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+        ['formSize', 'default | small', 'No', 'Controls form paddings and spacing. Use "small" to embed the form in small layout containers like a dropdown in the header.'],
+        ['hideCloseBtnOnEmailConfirmation', 'boolean', 'No', 'Controls the visibility of the “Close” button on the email confirmation view.'],
+        ['routeRedirectOnEmailConfirmationClose', 'function', 'No', 'Determines where the user is redirected when the “Close” button on the email confirmation view is clicked.'],
+        ['slots', 'function', 'No', 'Allows passing the SuccessNotification container or custom component rendered on successful sign-up if routeRedirectOnSignIn is not provided.'],
+        ['routeSignIn', 'function', 'No', 'Determines where the “Already a member? Sign in” link redirects the customer.'],
+        ['routeRedirectOnSignIn', 'function', 'No', 'Determines the page to which the user should be redirected after sign-in (if isAutoSignInEnabled is true). If provided, the user won’t see SuccessNotification or any override provided via slot.'],
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/sign-up.mdx]:
```suggestion
        ['routeRedirectOnSignIn', 'function', 'No', 'Determines the page to which the user should be redirected after sign-in (if isAutoSignInEnabled is true). If provided, the user won’t see SuccessNotification or any override provided through slot.'],
```

File: src/content/docs/dropins/user-auth/containers/success-notification.mdx [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+        ['slots', 'function', 'No', 'Allows overriding the SuccessNotification container action buttons (SuccessNotificationActions).'],
+        ['formSize', 'default | small', 'No', 'Controls form paddings and spacing. Use "small" to embed the form in small layout containers like a dropdown in the header.'],
+        ['headingText', 'string', 'No', 'Heading text content.'],
+        ['messageText', 'string', 'No', 'Message text content.'],
+    ]}
+/>
+
+## Example
+
+The following example renders the SignIn container. Upon success, the contain displays a welcome heading, a message, and two action buttons: "My Account" and "Logout".
```
Comment (jeff-matthews) [inline src/content/docs/dropins/user-auth/containers/success-notification.mdx]:
```suggestion
The following example renders the SignIn container. On success, the container displays a welcome heading, a message, and two action buttons: "My Account" and "Logout".
```

**Review by jeff-matthews — COMMENTED** [review]
Looks good! Just a few minor comments.

File: public/images/user-auth/reset-password-response.png
Comment (KostiantynFandeliuk) [inline public/images/user-auth/reset-password-response.png:1]: If this screenshot is intended for public use, I would suggest changing the email address on the screenshot.

File: public/images/user-auth/sign-in-check-email.png
Comment (KostiantynFandeliuk) [inline public/images/user-auth/sign-in-check-email.png:1]: If this screenshot is intended for public use, I would suggest changing the email address on the screenshot.

File: public/images/user-auth/sign-in-not-confirmed.png
Comment (KostiantynFandeliuk) [inline public/images/user-auth/sign-in-not-confirmed.png:1]: If this screenshot is intended for public use, I would suggest changing the email address on the screenshot.

File: public/images/user-auth/sign-up-verify-email.png
Comment (KostiantynFandeliuk) [inline public/images/user-auth/sign-up-verify-email.png:1]: If this screenshot is intended for public use, I would suggest changing the email address on the screenshot.

File: src/content/docs/dropins/user-auth/containers/sign-up.mdx [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+        ['onErrorCallback', 'function', 'No', ' Callback executed when an error occurs, receiving the error object as a parameter.'],
+        ['onSuccessCallback', 'function', 'No', 'Callback executed when the user successfully signs up, receiving userName and status as parameters.'],
```
Comment (KostiantynFandeliuk) [inline src/content/docs/dropins/user-auth/containers/sign-up.mdx:65]: Additionally, we have introduced a new prop responsible for the "Confirm Password" field.

`requireRetypePassword`

PR with screenshot - https://git.corp.adobe.com/Commerce-Storefront/storefront-auth/pull/58

**Review by jeff-matthews — APPROVED** [review]

**Review by KostiantynFandeliuk — APPROVED** [review]

**Pull request merged**